### PR TITLE
Firebase v3 + Linting

### LINKED ISSUE #23: Update to Firebase 3.x.x
Sad that the API has significant changes, but so things go. I've been using [re-base](https://github.com/tylermcginnis/re-base) but I love to play with your library. Do you have a plan for update?

### PR DESCRIPTION
**Pull request by prescottprue**
- Version 3+ working in Firebase (Fixes #23)
- No Dependency on Bluebird
- Removed `build` from git tracking
- Lint testing/fixing added through ESLint following JS-Standard
- Added the following NPM scripts:
  - `npm run lint` - Run Lint
  - `npm run lint:fix` - Run and Fix Lint
  - `npm run watch` - Build files using babel and rebuild if changed
  - `npm run clean` - Empty build folder before builds (used in other scripts)

### CODE REVIEW

**Comment by mrshll** [comment]
Working on testing this today! I'll post back my feedback.

File: source/compose.js [hunk 1]
```diff
@@ -1,39 +1,34 @@
-
 import Firebase from 'firebase'
 import * as Actions from './actions'
 
-
-
 export default (url, config) => {
```
Comment (mrshll) [inline source/compose.js]: first and only argument now is `config`

File: source/compose.js [hunk 2]
```diff
@@ -1,39 +1,34 @@
...
     const defaultConfig = {
       userProfile: null
     }
-
     const store = next(reducer, initialState)
 
     const {dispatch} = store
+    // console.log('firebase:', Firebase, Object.getOwnPropertyNames(Firebase))
+    try {
+      Firebase.initializeApp(url)
```
Comment (mrshll) [inline source/compose.js]: `Firebase.initializeApp(config)`

File: source/compose.js [hunk 3]
```diff
@@ -1,39 +1,34 @@
...
+    } catch (err) { console.warn('Firebase error:', err) }
 
-    const ref = new Firebase(url)
+    const ref = Firebase.database()
```
Comment (mrshll) [inline source/compose.js]: `const ref = Firebase.database().ref()`

File: source/compose.js [hunk 4]
```diff
@@ -1,39 +1,34 @@
...
-    }
-
+    const firebase = Object.assign({}, ref, Firebase, { _: {
+      watchers: {},
+      config: configs,
+      authUid: null
+    }})
 
     const set = (path, value, onComplete) => ref.child(path).set(value, onComplete)
     const push = (path, value, onComplete) => ref.child(path).push(value, onComplete)
```
Comment (mrshll) [inline source/compose.js:36]: do the above or change these `child`'s to `ref`

**Comment by mrshll** [comment]
Found some problems - it looks like you cannot take the child of the instance returned by `Database()` which prevents actions on children.

**Comment by prescottprue** [comment]
@mmoutenot Can you see if you are seeing these problems on [redux-firebasev3](https://github.com/prescottprue/redux-firebasev3)?

**Comment by mrshll** [comment]
Sure thing. Working off my own redux-react-firebase branch now and need to finish things today -- I'll try to do tonight or tomorrow morning.

**Pull request merged**